exchanges/okx/okx_websocket.go

```diff
@@ -1708,6 +1682,12 @@ func (m *wsRequestDataChannelsMultiplexer) Run() {
 	}
 }
 
+// Shutdown causes the multiplexer to exit its Run loop
+// All channels are left open, but websocket shutdown first will ensure no more messages block on multiplexer reading
+func (m *wsRequestDataChannelsMultiplexer) Shutdown() {
+	close(m.shutdown)
```

Comment:
I don't agree that's safe to do, because Run() is likely to read from the NEW m.shutdown.

However we could remedy that by putting a m.shutdown = make(chan bool) in Run, right?

Reply:
Option 1: Localise a var to the shutdown chan in Run before run starts. Still has risk of race though.
Option 2: Have Run re-make the chan itself after consuming the close. Only limitation: Single consumer pattern. Anything else watching shutdown would get scuppered. I think that's safe given size and scope of wsRequestDataChannelsMultiplexer

So I'll fix with Option 2